examples/main/main.cpp

```diff
@@ -331,6 +334,17 @@ int main(int argc, char ** argv) {
     while ((n_remain != 0 && !is_antiprompt) || params.interactive) {
         // predict
         if (embd.size() > 0) {
+            auto max_embd_size = n_ctx - 4;
+            // Ensure the input doesn't exceed the context size by truncating embd if necessary.
+            if ((int)embd.size() > max_embd_size) {
+                auto skipped_tokens = embd.size() - max_embd_size;
+                console_set_color(con_st, CONSOLE_COLOR_ERROR);
+                printf("<<input too long: skipped %ld token%s>>", skipped_tokens, skipped_tokens != 1 ? "s" : "");
```

Comment:
Might be a good idea to print this to `stderr` instead, to keep the output of the model clean in stdout.

Reply:
This is something that only would apply in interactive mode. It also would be a little weird to do that since `console_set_color` will send the color codes to stdout anyway.